src/main/scala/coreplex/BaseCoreplex.scala

```diff
@@ -110,14 +110,12 @@ abstract class BaseCoreplexModule[+L <: BaseCoreplex, +B <: BaseCoreplexBundle](
     l1tol2net.io.managers <> managerEndpoints.map(_.innerTL) :+ mmioManager.io.inner
 
     // Create a converter between TileLinkIO and MemIO for each channel
-    val mem_ic = Module(new TileLinkMemoryInterconnect(nBanksPerMemChannel, c.nMemChannels)(outermostParams))
+    val mem_ic = Module(new TileLinkMemoryInterconnect(nBanksPerMemChannel, c.nMemChannels)(outerMemParams))
 
-    val outerTLParams = p.alterPartial({ case TLId => "L2toMC" })
     val backendBuffering = TileLinkDepths(0,0,0,0,0)
     for ((bank, icPort) <- managerEndpoints zip mem_ic.io.in) {
       val enqueued = TileLinkEnqueuer(bank.outerTL, backendBuffering)
-      val unwrapped = TileLinkIOUnwrapper(enqueued)
-      TileLinkWidthAdapter(icPort, unwrapped)
+      icPort <> TileLinkIOUnwrapper(enqueued)
```

Comment:
I see what you're doing here now.  You've moved the width conversion out to the Periphery.  I support this change.